cocos/platform/openharmony/OpenHarmonyPlatform.cpp

```diff
@@ -0,0 +1,335 @@
+                CC_ASSERT(nativexcomponet != nullptr);        
+                platform->onSurfaceChanged(nativexcomponet, msgData.window);
+            } else if (msgData.type == MessageType::WM_XCOMPONENT_SURFACE_DESTROY) {
+                OH_NativeXComponent* nativexcomponet = reinterpret_cast<OH_NativeXComponent*>(msgData.data);
+                CC_ASSERT(nativexcomponet != nullptr);            
+                platform->onSurfaceDestroyed(nativexcomponet, msgData.window);
+            } else {
+                CC_ASSERT(false);
+            }
+            continue;
+        }
+
+        if (msgData.type == MessageType::WM_APP_SHOW) {
+            platform->onShowNative();
+        } else if (msgData.type == MessageType::WM_APP_HIDE) {
+            platform->onHideNative();
+        } else if (msgData.type == MessageType::WM_APP_DESTROY) {
+            platform->onDestroyNative();
+        }
+        // if(msgData.type == MessageType::WM_VSYNC) {
+        //     platform->runTask();
+        // }
+    }
+}
+
+void OpenHarmonyPlatform::onCreateNative(napi_env env, uv_loop_t* loop) {
+    LOGD("OpenHarmonyPlatform::onCreateNative");
+}
+
+void OpenHarmonyPlatform::onShowNative() {
+    LOGD("OpenHarmonyPlatform::onShowNative");
+    EventDispatcher::dispatchOnResumeEvent();
+}
+
+void OpenHarmonyPlatform::onHideNative() {
+    LOGD("OpenHarmonyPlatform::onHideNative");
+    EventDispatcher::dispatchOnPauseEvent();
+}
+
+void OpenHarmonyPlatform::onDestroyNative() {
+    LOGD("OpenHarmonyPlatform::onDestroyNative");
+}
+
+void OpenHarmonyPlatform::timerCb(uv_timer_t* handle) {
+    if(OpenHarmonyPlatform::getInstance()->eglCore_ != nullptr){
+        OpenHarmonyPlatform::getInstance()->tick();
+        OpenHarmonyPlatform::getInstance()->eglCore_->Update();
+    }
+}
+
+void OpenHarmonyPlatform::workerInit(napi_env env, uv_loop_t* loop) {
+    _workerLoop = loop;
+    if (_workerLoop) {
+        uv_async_init(_workerLoop, &_messageSignal, reinterpret_cast<uv_async_cb>(OpenHarmonyPlatform::onMessageCallback));
+        if (!_messageQueue.empty()) {
+            triggerMessageSignal(); // trigger the signal to handle the pending message
+        }
+    }
+}
+
+void OpenHarmonyPlatform::requestVSync() {
+    // OH_NativeVSync_RequestFrame(OpenHarmonyPlatform::getInstance()->_nativeVSync, OnVSync, nullptr);
+     if (_workerLoop) {
+        // Todo: Starting the timer in this way is inaccurate and will be fixed later.
+        uv_timer_init(_workerLoop, &_timerHandle);
+        // The tick function needs to be called as quickly as possible because it is controlling the frame rate inside the engine.
+        uv_timer_start(&_timerHandle, &OpenHarmonyPlatform::timerCb, 0, 1);
+     }
+}
+
+int32_t OpenHarmonyPlatform::loop() {
+    return 0;
+}
+
+void OpenHarmonyPlatform::onSurfaceCreated(OH_NativeXComponent* component, void* window) {
+    eglCore_ = new EGLCore();
```

Comment:
Why do we need to use EGLCore?

Reply:
EGLCore is new class that handle egl logic,not java eglcore